test/suite/test_txn06.py

```diff
@@ -44,12 +44,17 @@ def test_long_running(self):
         # Populate a table
         SimpleDataSet(self, self.source_uri, self.nrows).populate()
 
-        # Now scan the table and copy the rows into a new table
+        # Now scan the table and copy the rows into a new table. The cursor will keep the snapshot
```

Comment:
Wouldn't you need to do `self.session.begin_transaction()` to pin a snapshot?

Reply:
No, keeping a cursor positioned or scanning (with next/prev) keeps a snapshot pinned.  The problem before was the same session was doing the inserts, and the inserts caused the snapshot to be updated because each one is an auto-commit transaction.

That said, if it would be clearer with an explicit `begin_transaction`, I'm happy to add that.